### PR TITLE
Add some useful (internal) helpers

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: libchisel/src/utils.rs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+pub trait SerializationHelpers {
+    /// Deserialize bytecode to a Module.
+    fn from_slice(input: &[u8]) -> Module;
+    /// Serialize Module to bytecode. Serialization consumes the input.
+    fn to_vec(self) -> Vec<u8>;
+}
+
+impl SerializationHelpers for Module {
+    fn from_slice(input: &[u8]) -> Self {
+        deserialize_buffer::<Module>(&input).expect("valid Wasm bytecode")
```
Comment (jakelang) [inline libchisel/src/utils.rs]: On failure, should this not be `invalid Wasm bytecode`? same for below.

File: libchisel/src/utils.rs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+pub trait HasNamesSection {
+    /// Returns true if the module has a NamesSection.
+    fn has_names_section(&self) -> bool;
+}
+
+impl HasNamesSection for Module {
+    fn has_names_section(&self) -> bool {
+        // Lets forcefully parse in case it wasn't yet.
+        let mut module = self.clone().parse_names().unwrap();
```
Comment (jakelang) [inline libchisel/src/utils.rs]: I would prefer an `expect()` here rather than unwrap.

File: libchisel/src/utils.rs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        module.names_section().is_some()
+    }
+}
+
+pub trait SerializationHelpers {
+    /// Deserialize bytecode to a Module.
+    fn from_slice(input: &[u8]) -> Module;
```
Comment (jakelang) [inline libchisel/src/utils.rs]: Are you sure we would rather panic internally rather than return a `Result<T>`?

File: libchisel/src/utils.rs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    /// Serialize Module to bytecode. Serialization consumes the input.
+    fn to_vec(self) -> Vec<u8>;
```
Comment (jakelang) [inline libchisel/src/utils.rs]: Same as above.

File: libchisel/src/utils.rs (hunk 1 above)
Comment (axic) [reply libchisel/src/utils.rs]: Dunno, "expect valid wasm bytecode" sounded nice.

File: libchisel/src/utils.rs (hunk 2 above)
Comment (axic) [reply libchisel/src/utils.rs]: Yeah fair, forgot to change it amidst copy pastes.

File: libchisel/src/utils.rs (hunk 3 above)
Comment (axic) [reply libchisel/src/utils.rs]: That was considered as an option, but this right now is only using in testing code, so its fine.

File: libchisel/src/utils.rs (hunk 1 above)
Comment (jakelang) [reply libchisel/src/utils.rs]: I think it makes sense to do "Invalid Wasm bytecode/module" as the panic message does not say "Expected x". may cause confusion.

File: libchisel/src/utils.rs (hunk 3 above)
Comment (jakelang) [reply libchisel/src/utils.rs]: ok

**Pull request merged**